pkg/cloud/user_credentials_test.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud_test
+
+import (
+	"github.com/aws/cluster-api-provider-cloudstack/pkg/cloud"
+	"github.com/aws/cluster-api-provider-cloudstack/test/dummies"
+	. "github.com/onsi/ginkgo"
+	"github.com/pkg/errors"
+
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("User Credentials", func() {
+
+	BeforeEach(func() {
+		dummies.SetDummyVars()
+		dummies.SetDummyClusterStatus()
+		dummies.SetDummyCSMachineStatuses()
+	})
+
+	AfterEach(func() {
+	})
+
+	Context("UserCred Semi-Integ Tests", func() {
+		client, connectionErr := cloud.NewClient("../../cloud-config")
```

Comment:
should we be hardcoding this path in the tests? Doesn't it assume a user has the file there when they might not?

Reply:
Completely, we should not.

This and other issues are why I want to turn my attention towards our CI/CD this week.

The semi-integ unit tests only work locally at the moment. They do pass for me, but there are other configuration hurdles besides this one that is preventing the semi-integs from working. For example, the name of the sub domain user should be configurable.

The E2E tests have some of this configuration, but not enough to run all tests. So, yeah. Working to get to this this week. : )